src/util.js

```diff
@@ -0,0 +1,124 @@
+var https   = require('https');
+var http   = require('http');
+
+var Buffer  = require("buffer").Buffer;
+var log     = require('npmlog');
+var mime    = require('mime');
+
+/**
+  Utility module for regularly used functions.
+*/
+
+/**
+ * downloadFile - This function will take a URL, upload it to Matrix and return
+ * the corresponding MXC url in a Promise. The content will be uploaded on the
+ * users behalf using the ID, or the AS bot if set to null.
+ *
+ * @param  {string} url      The URL to be downloaded from.
+ * @return {Promise<Buffer>} A promise that will return a buffer with the data.
+ */
+function downloadFile(url){
+  return new Promise((resolve, reject) => {
+
+    if(url.startsWith("https")){
+      var ht = https;
+    }
+    else {
+      var ht = http;
+    }
+
+    ht.get((url), (res) => {
+        var size = parseInt(res.headers["content-length"]);
+        var buffer = Buffer.alloc(size);
```

Comment:
>You *really* sure you want to set your min version to 5+?

``` js
if(Buffer.alloc) {//Since 5.10
   buffer = Buffer.alloc(size);
}
else {//Deprecated
   buffer = new Buffer(size);
}
```

How about that?